### PR TITLE
Refactor/sync case completions

### PR DESCRIPTION
**Pull request by dannil76**
## Explain the changes you’ve made
Refactored to new code style implementation

Added tests

## How to test
1. Checkout this branch
2. run `yarn test --verbose` within service folder for `service viva-ms`

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-2np9j3a Skriv om SyncCaseCompletions till överenskommet format](https://app.clickup.com/t/2np9j3a)

File: services/viva/microservice/test/lambdas/syncCaseCompletions.test.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  const updateCaseMock = jest.fn();
+  return {
+    putSuccessEvent: () => Promise.resolve(),
+    updateCase: updateCaseMock,
+    validateStatusCode: () => statusCode,
+    getLatestWorkflowId: () => Promise.resolve(workflowId),
+    getWorkflowCompletions: () => Promise.resolve(vadaCompletions),
+    getCaseOnWorkflowId: () => Promise.resolve(caseToUpdate),
+  };
+}
```
Comment (kajm) [inline services/viva/microservice/test/lambdas/syncCaseCompletions.test.ts:112]: By adding the `updateCaseMock` here we cannot alter the implementation in individual tests if wanted.

File: services/viva/microservice/test/lambdas/syncCaseCompletions.test.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+it('updates a case with new completion information received from Viva', async () => {
+  const dependencies = createDependencies(false);
+  const result = await syncCaseCompletions(input, dependencies);
+  expect(result).toBe(true);
+  expect(dependencies.updateCase).toHaveBeenCalledWith(
+    caseKeys,
+    expect.objectContaining(vadaCompletions)
+  );
```
Comment (kajm) [inline services/viva/microservice/test/lambdas/syncCaseCompletions.test.ts:121]: Add newline

File: services/viva/microservice/test/lambdas/syncCaseCompletions.test.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+});
+
+it('does nothing if Viva status code is 1', async () => {
+  const dependencies = createDependencies(true);
+  const result = await syncCaseCompletions(input, dependencies);
```
Comment (kajm) [inline services/viva/microservice/test/lambdas/syncCaseCompletions.test.ts:126]: add newline

**Review by kajm — APPROVED** [review]

File: services/viva/helpers/createCase.ts [hunk 4]
```diff
@@ -37,8 +37,8 @@ function stripNonNumericalCharacters(valueIncludingChars: string) {
 
 function getPeriodInMilliseconds(vivaApplication: VivaMyPagesPersonApplication): CasePeriod {
   return {
-    startDate: Date.parse(vivaApplication.period.start),
-    endDate: Date.parse(vivaApplication.period.end),
+    startDate: Date.parse(`${vivaApplication.period.start}T00:00:00.000Z`),
+    endDate: Date.parse(`${vivaApplication.period.end}T00:00:00.000Z`),
   };
 }
```
[the remark this answers is not in the document]
Comment (dannil76) [reply services/viva/helpers/createCase.ts:44]: Will keep current implementation due to this PR is overdue and it works.

**Pull request merged**